### DISCUSSION TITLE
classes/class_menubar

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#37] [User-contributed notes]
# classes/class_menubar

Inherits: Control< CanvasItem< Node< Object A horizontal menu bar that creates a MenuButton for each PopupMenu child. Description: A horizontal menu bar that creates a MenuButton for each PopupMenu...

https://docs.godotengine.org/en/latest/classes/class_menubar.html

**Comment by Coffee7993** [comment]
`prefer_global_menu` seems to be a feature specific to just MacOS.

Sources:
- [Godot forum post](https://forum.godotengine.org/t/prefer-global-menu-in-menubar/36638)
- [GitHub PR](https://github.com/godotengine/godot/pull/63950)

**Reply by Calinou** [reply]
Indeed, global menus are currently only implemented on macOS. Linux desktop environments sometimes have global menus but they're uncommon, and the API to support them isn't widely used in third-party applications.

Windows doesn't have a concept of a global menu in the first place.

**Comment by ScatteredComet** [comment]
I assume that, under `prefer_global_menu`, its meant to read

Note: If true and global menu is _not_ supported, this node is not displayed, has zero size, and all its child nodes except PopupMenus are inaccessible.

If someone can confirm this then I'll happily submit a PR.

**Reply by Calinou** [reply]
The current description is correct. The property only has an effect if the current platform supports global menus. Otherwise, it will do nothing.

When using a global menu, Godot doesn't display the menu bar, but the OS does instead. All the rendering is performed by the OS, which means the node is effectively invisible from Godot's perspective.

**Reply by ScatteredComet** [reply]
Oops! I see what you mean. Sorry about that.
